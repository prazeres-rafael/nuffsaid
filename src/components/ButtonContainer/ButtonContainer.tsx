import { Button } from "@mui/material";
import React from "react";

type Props = {
  getMessages: boolean;
  handleStop: () => void;
  handleClear: () => void;
};

export default function ButtonContainer({
  getMessages,
  handleStop,
  handleClear,
}: Props): React.ReactElement {
  return (
    <>
      <Button onClick={handleStop} variant="contained" color="success">
        {getMessages ? "STOP" : "START"}
      </Button>
      <Button onClick={handleClear} color="success">
        CLEAR
      </Button>
    </>
  );
}
