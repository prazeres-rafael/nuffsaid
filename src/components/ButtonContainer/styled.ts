import styled from "styled-components";
import { palette } from "../../utils/palette";

export const Button = styled.button({
  backgroundColor: palette.messages.info,
  cursor: "pointer",
  margin: 5,
  padding: "12px 25px 12px 25px",
  borderRadius: 5,
  borderWidth: 0,

  "&:hover": {
    backgroundColor: "#16F2AA",
  },
});
