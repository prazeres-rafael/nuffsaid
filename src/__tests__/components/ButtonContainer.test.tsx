import { render } from "@testing-library/react";
import ButtonContainer from "../../components/ButtonContainer/ButtonContainer";

describe("ButtonContainer", () => {
  it("should match the snapshot", () => {
    const { container } = render(
      <ButtonContainer
        getMessages={true}
        handleStop={jest.fn()}
        handleClear={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
