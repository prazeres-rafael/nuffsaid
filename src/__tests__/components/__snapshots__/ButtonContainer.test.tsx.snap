// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should match the snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ke5b6m-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    STOP
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1lg55su-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    CLEAR
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
